tests/test_processing_rt.py

```diff
@@ -0,0 +1,258 @@
+        np.array(
+            [
+                [0.5773502691896257, 0.7071067811865475, -0.408248290463863, 1.0],
+                [0.5773502691896257, 0.0, 0.816496580927726, 2.0],
+                [0.5773502691896257, -0.7071067811865475, -0.408248290463863, 3.0],
+                [0, 0, 0, 1.0],
+            ]
+        )
+    )
+
+    np.testing.assert_array_equal(rt_xy_from_known_m, rt_xy_expected)
+
+    exception_default_params = dict(
+        origin=all_m.isel(channel=[0]),
+        axis_1=all_m.isel(channel=[0, 1]),
+        axis_2=all_m.isel(channel=[0, 2]),
+        axes_name="xy",
+        axis_to_recalculate="y",
+    )
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(origin=all_m.isel(channel=[0, 1]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_1=all_m.isel(channel=[0]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_2=all_m.isel(channel=[0]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{
+                **exception_default_params,
+                **dict(axis_1=all_m.isel(channel=[0, 1], time=slice(None, 50))),
+            }
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="yyz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="xxz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="zzz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_to_recalculate="h")}
+        )
+
+
+def test_rt_transpose():
+    n_frames = 10
+    angles = Angles.from_random_data(size=(3, 1, n_frames))
+    rt = Rototrans.from_euler_angles(angles, angle_sequence="xyz")
+
+    rt_t = Rototrans.from_transposed_rototrans(rt)
+
+    rt_t_expected = np.zeros((4, 4, n_frames))
+    rt_t_expected[3, 3, :] = 1
+    for row in range(rt.row.size):
+        for col in range(rt.col.size):
+            for frame in range(rt.time.size):
+                rt_t_expected[col, row, frame] = rt[row, col, frame]
+
+    for frame in range(rt.time.size):
+        rt_t_expected[:3, 3, frame] = -rt_t_expected[:3, :3, frame].dot(
+            rt[:3, 3, frame]
+        )
+
+    np.testing.assert_array_almost_equal(rt_t, rt_t_expected, decimal=10)
+
+
+def test_average_rt():
+    # TODO: investigate why this does not work
+    # angles = Angles.from_random_data(size=(3, 1, 100))
+    # or
+    # angles = Angles(np.arange(300).reshape((3, 1, 100)))
+    angles = Angles(np.random.rand(3, 1, 100))
+    seq = "xyz"
+
+    rt = Rototrans.from_euler_angles(angles, seq)
```

Comment:
from_random

Reply:
I tried, but I'm getting some strange behavior.

Optimization works with that:

```python
angles = Angles(np.random.rand(3, 1, 100))
```

But not that:

```python
angles = Angles.from_random_data()
```

In pyomeca v2, `from_random_data` is the equivalent to this:

```python
angles = Angles(np.random.rand(3, 1, 100).cumsum(-1))
```

I don't see any reason why that shouldn't work.
Which leads me to believe that we have a bug...
Looking quickly at [our implementation](https://github.com/romainmartinez/pyomeca/blob/f7f770c1d9ceab0cf8235fab8db4dfb253d8fb62/pyomeca/processing/rototrans.py#L181-L198), do you see anything wrong?

Reply:
Can you open an issue for that with a minimal code that fails?
I'll have a look!

Reply:
Issue #91 opened!